worker/uniter/remotestate/watcher.go

```diff
@@ -877,10 +899,39 @@ func (w *RemoteStateWatcher) leaderSettingsChanged() error {
 	return nil
 }
 
-func (w *RemoteStateWatcher) leadershipChanged(isLeader bool) {
+func (w *RemoteStateWatcher) leadershipChanged(isLeader bool) error {
 	w.mu.Lock()
+	defer w.mu.Unlock()
+
 	w.current.Leader = isLeader
-	w.mu.Unlock()
+	if !isLeader {
+		if w.secretRotateWatcher != nil {
+			_ = worker.Stop(w.secretRotateWatcher)
+		}
+		w.secretRotateWatcher = nil
+		w.rotateSecretsChanges = nil
```

Comment:
should this channel be closed?

Reply:
no because the sender is the worker it is handed off to and it could result in a panic
the worker is shut down separately